apps/prairielearn/src/pages/instructorStudents/instructorStudents.html.tsx

```diff
@@ -0,0 +1,226 @@
+      columnHelper.accessor('user.uid', {
+        header: 'UID',
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor('user.name', {
+        header: 'Name',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('user.email', {
+        header: 'Email',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('enrollment.created_at', {
+        header: `Enrolled on (${formatTz(timezone)})`,
+        cell: (info) => {
+          const date = new Date(info.getValue());
+          return (
+            <div style={{ fontVariantNumeric: 'tabular-nums' }}>
+              {formatDate(date, timezone, {
+                includeTz: false,
+              })}
+            </div>
+          );
+        },
+      }),
+    ],
+    [],
+  );
+
+  const allColumnIds = columns
+    .filter((col) => col.accessorKey)
+    .map((col) => col.accessorKey.replace('.', '_'));
+  const [columnVisibility, setColumnVisibility] = useQueryState(
+    'columns',
+    parseAsColumnVisibilityStateWithColumns(allColumnIds).withDefault(
+      Object.fromEntries(allColumnIds.map((id) => [id, true])),
+    ),
+  );
+
+  const table = useReactTable({
+    data: students,
+    columns,
+    columnResizeMode: 'onChange',
+    getRowId: (row) => row.user.user_id,
+    state: {
+      sorting,
+      columnFilters,
+      globalFilter,
+      columnSizing,
+      columnVisibility,
+      columnPinning,
+    },
+    onSortingChange: setSorting,
+    onColumnFiltersChange: setColumnFilters,
+    onGlobalFilterChange: setGlobalFilter,
+    onColumnSizingChange: setColumnSizing,
+    onColumnVisibilityChange: setColumnVisibility,
+    onColumnPinningChange: setColumnPinning,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    defaultColumn: {
+      minSize: 150,
+      size: 300,
+      maxSize: 500,
+      enableSorting: true,
+      enableHiding: true,
+      enablePinning: true,
+    },
+  });
+
+  return (
+    <div class="card d-flex flex-column h-100">
+      <div class="card-header bg-primary text-white">
+        <div class="d-flex align-items-center justify-content-between">
+          <div>Students</div>
+          <div>
+            <DownloadButton
+              students={students}
+              course={course}
+              courseInstance={courseInstance}
+              table={table}
+            />
+          </div>
+        </div>
+      </div>
+      <div class="card-body d-flex flex-column">
+        <div class="d-flex flex-row flex-wrap align-items-center mb-3 gap-2">
+          <div class="flex-grow-1">
+            <div class="input-group">
+              <input
+                ref={searchInputRef}
+                type="text"
+                class="form-control"
+                aria-label="Search by UID, name or email."
+                placeholder="Search by UID, name, email..."
```

Comment:
The search box is now very wide, so the clear button is not obviously connected to the content. Could we revert to the original search box size?

Reply:
https://github.com/user-attachments/assets/5f7b0b84-0e53-4824-97a0-dd8f1fb15541

Reply:
(and 7de708f99035d8d34fd7834f9f5ddf3255a5324e)

Reply:
Sorry Matt! I'd thought this would be an uncontroversial change so I just committed and pushed it. This was a direct copy of the way GitHub organizes its pages, with the search bar spanning all available width. It also matched our own Issues page.

IMO I'd still prefer that. The current layout doesn't make good use of space on narrower viewports.

<img width="881" alt="Screenshot 2025-06-27 at 09 06 55" src="https://github.com/user-attachments/assets/8bfdc548-7893-40c5-a034-adc64b55d754" />

Reply:
I removed the breakpoint for `md` in 17ea79c797ba90406af4c912536d8ab2349f7d24, which in my testing made both behaviors (very wide screens and very narrow screens) work well.